addons/sale_pdf_quote_builder/static/src/js/quotation_document_kanban/quotation_document_kanban_controller.js

```diff
@@ -11,3 +14,19 @@ export class QuotationDocumentKanbanController extends KanbanController {
         this.uploadRoute = '/sale_pdf_quote_builder/quotation_document/upload';
     }
 }
+
+patch(UploadButton.prototype, {
+    setup() {
+        super.setup();
+        this.notification = useService('notification');
+    },
+    async onFileInputChange(ev) {
+        if ([...ev.target.files].some(file => !file.name.endsWith('.pdf'))) {
```

Comment:
IMO this isn't really reliable, customers that rename pictures with ".pdf" at the end would still end up having the traceback. I'd rely on the MIME type only to check the type of the file